apps/storefront/src/pages/QuoteDetail/index.tsx

```diff
@@ -228,7 +238,25 @@ function QuoteDetail() {
     });
   }, [isB2BUser, quoteDetail, selectCompanyHierarchyId, purchasabilityPermission]);
 
-  useEffect(() => {
+  const validateBackendProducts = async () => {
+    setIsValidationLoading(true);
+    const { errors } = await validateProducts(productList);
+
+    if (errors.length) {
+      errors.forEach((error) => {
+        if (error.type === 'validation') {
+          snackbar.error(error.message);
+        }
+      });
+
+      setShouldHideCheckoutButton(true);
+    }
+
+    setIsHideQuoteCheckout(false);
```

Comment:
these seem to be very similar - any reason we couldn't rely on the existing variable?

Reply:
i added a new state variable because the current `isHideQuoteCheckout` blocks the checkout to be performed if there's any error in a product but doesn't actually disabled to button or anything. If i use the same one i would end up disabling the button even when only using front end validations, and i didn't want change anything from the current behavior. I agree that it does feels duplicated but is easier to manage some states separately for the different flows.